File: spring-chat/src/main/java/com/example/springchat/model/MessageRequest.java
package com.example.springchat.model;

public class MessageRequest {

    private Long recipientId;
    private String content;

    private Long messageId;
    private Boolean updateMessage = false;


    public Long getRecipientId() {
        return recipientId;
    }

    public void setRecipientId(Long recipientId) {
        this.recipientId = recipientId;
    }

    public String getContent() {
        return content;
    }

    public void setContent(String content) {
        this.content = content;
    }

    public Long getMessageId() {
        return messageId;
    }

    public void setMessageId(Long messageId) {
        this.messageId = messageId;
    }

    public Boolean getUpdateMessage() {
        return updateMessage;
    }

    public void setUpdateMessage(Boolean updateMessage) {
        this.updateMessage = updateMessage;
    }
}

File: spring-chat-client/src/app/_dtos/chat/MessageRequest.ts
export class MessageRequest {
  recipientId: number
  content: string
  messageId: number
  chatType: string
  chatId: number

  constructor(recipientId: number, content: string, messageId: number, chatType: string, chatId: number) {
    this.recipientId = recipientId;
    this.content = content;
    this.messageId = messageId;
    this.chatType = chatType;
  }
}
